cmd/util/repo.go

```diff
@@ -35,3 +37,83 @@ func SetupRepoConfig(cmd *cobra.Command) (types.BacalhauConfig, error) {
 
 	return bacalhauCfg, nil
 }
+
+func SetupConfig() (config.ReadWriter, error) {
+	// get the global viper instance
+	v := viper.GetViper()
+	// get the repo path set in the root command, and ensure it's set (it will always be set unless dev error)
+	repoPath := v.GetString("repo")
+	if repoPath == "" {
+		return nil, fmt.Errorf("repo path not set")
+	}
+	// check if the user specified config files via the --config flag
+	configFiles := v.GetStringSlice(cliflags.RootCommandConfigFiles)
+	if len(configFiles) == 0 {
+		// if no files were provided we look in repo and xdg config, the latter takes precedence
+		repoConfig, err := getConfigFileAtPath(repoPath)
+		if err != nil {
+			return nil, err
+		}
+		var xdgConfig string
+		xdgPath, err := os.UserConfigDir()
+		if err != nil {
+			log.Info().Err(err).Msg("find to find user config dir")
+		} else {
+			xdgConfig, err = getConfigFileAtPath(xdgPath)
+			if err != nil {
+				return nil, err
+			}
+		}
+		configFiles = append(configFiles, repoConfig, xdgConfig)
```

Comment:
you should look for `xddConfig` and only if not preset (indicating no migration yet), then look at `repoConfig`